src/database/remove.ts

```diff
@@ -0,0 +1,98 @@
+import * as pathLib from "path";
+import { Queue } from "../queue";
+import * as FirebaseError from "../error";
+import * as logger from "../logger";
+import RemoveRemote, { PrefetchResult } from "./remove-remote";
+
+export interface DatabaseRemoveOptions {
+  // RTBD instance ID.
+  instance: string;
+  // Number of concurrent chunk deletes to allow.
+  concurrency: number;
+  // Number of retries for each chunk delete.
+  retries: number;
+}
+
+export default class DatabaseRemove {
+  public path: string;
+  public concurrency: number;
+  public retries: number;
+  public remote: any;
+  private jobQueue: Queue<string>;
+  private waitingPath: Map<string, number>;
+
+  /**
+   * Construct a new RTDB delete operation.
+   *
+   * @constructor
+   * @param path path to delete.
+   * @param the RTDB instance ID.
+   * @param options.concurrency the number of concurrent chunk delete allowed
+   * @param options.retires the number of retries for each chunk delete
+   */
+  constructor(path: string, options: DatabaseRemoveOptions) {
+    this.path = path;
+    this.concurrency = options.concurrency;
+    this.retries = options.retries;
+    this.remote = new RemoveRemote(options.instance);
+    this.waitingPath = new Map();
+    this.jobQueue = new Queue({
+      name: "long delete queue",
+      concurrency: this.concurrency,
+      handler: this.chunkedDelete.bind(this),
+      retries: this.retries,
+    });
+  }
+
+  public execute(): Promise<void> {
+    const prom: Promise<void> = this.jobQueue.wait();
+    this.jobQueue.add(this.path);
+    return prom;
+  }
+
+  private chunkedDelete(path: string): Promise<any> {
+    return this.remote
+      .prefetchTest(path)
+      .then((test: PrefetchResult) => {
+        switch (test) {
```

Comment:
this should have a `default` case simply as a best practice to catch any errors that might slip through (e.g. if a new value is added to `PrefetchResult` and not handled here).

Reply:
Sure.
Just fixed it. Let me know if there are anything else.